dask_sql/physical/utils/statistics.py

```diff
@@ -0,0 +1,207 @@
+
+            ``{'name': 'col0', 'min': 0, 'max': 100}``
+    """
+
+    # Check that we have a supported `ddf` object
+    if not isinstance(ddf, dd.DataFrame):
+        raise ValueError(f"Expected Dask DataFrame, got {type(ddf)}.")
+
+    # Be strict about columns argument
+    if columns:
+        if not isinstance(columns, list):
+            raise ValueError(f"Expected list, got columns={type(columns)}.")
+        elif not set(columns).issubset(set(ddf.columns)):
+            raise ValueError(f"columns={columns} must be a subset of {ddf.columns}")
+
+    # Extract "read-parquet" layer from ddf
+    try:
+        layer = hlg_layer(ddf.dask, "read-parquet")
+    except KeyError:
+        layer = None
+
+    # Make sure we are dealing with a
+    # ParquetFunctionWrapper-based DataFrameIOLayer
+    if not isinstance(layer, DataFrameIOLayer) or not isinstance(
+        layer.io_func, ParquetFunctionWrapper
+    ):
+        logger.warning(
+            f"Could not extract Parquet statistics from {ddf}."
+            f"\nAttempted IO layer: {layer}"
+        )
+        return None
+
+    # Collect statistics using layer information
+    parts = layer.inputs
+    fs = layer.io_func.fs
+    engine = layer.io_func.engine
+    if not isinstance(engine(), ArrowDatasetEngine):
+        logger.warning(
+            f"Could not extract Parquet statistics from {ddf}."
+            f"\nUnsupported parquet engine: {engine}"
+        )
+        return None
+
+    # Set default
+    if parallel is None:
+        parallel = False if _is_local_fs(fs) else 16
+    parallel = int(parallel)
+
+    if parallel:
+        # Group parts corresponding to the same file.
+        # A single task should always parse statistics
+        # for all these parts at once (since they will
+        # all be in the same footer)
+        groups = defaultdict(list)
+        for part in parts:
+            path = part.get("piece")[0]
+            groups[path].append(part)
+        group_keys = list(groups.keys())
+
+        # Compute and return flattened result
+        func = delayed(_read_partition_stats_group)
+        result = dask.compute(
+            [
+                func(
+                    list(
+                        itertools.chain(
+                            *[groups[k] for k in group_keys[i : i + parallel]]
+                        )
+                    ),
+                    fs,
+                    engine,
+                    columns=columns,
+                )
+                for i in range(0, len(group_keys), parallel)
+            ],
+            **(compute_kwargs or {}),
+        )[0]
+        return list(itertools.chain(*result))
+    else:
+        # Serial computation on client
+        return _read_partition_stats_group(parts, fs, engine, columns=columns)
+
+
+@make_pickable_without_dask_sql
+def _read_partition_stats_group(parts, fs, engine, columns=None):
+    def _read_partition_stats(part, fs, columns=None):
```

Comment:
@charlesbluca - The `make_pickable_without_dask_sql` decorator seems to do the trick here

Reply:
Thanks for looking into this @rjzamora 😄